cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1277,6 +1285,25 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                                 }
 
                                 this.kafkaExternalBootstrapDnsName.add(bootstrapAddress);
+
+                                Kafka resource = crdOperator.get(namespace, name);
+                                KafkaStatus ks = resource.getStatus();
+                                ListenersStatus ls = new ListenersStatus();
+
+                                if (ks == null) {
+                                    ks = new KafkaStatus();
+                                }
+
+                                if (ks.getListeners() != null) {
+                                    ls.setPlain(ks.getListeners().getPlain());
+                                    ls.setTls(ks.getListeners().getTls());
+                                }
+                                String externalBootstrapHost = serviceName + "." + namespace + ".svc";
+                                ls.setExternal(new ListenerStatus(externalBootstrapHost, kafkaCluster.getExternalPort(), null));
+                                ks.setListeners(ls);
+
+                                Kafka resourceWithNewStatus = new KafkaBuilder(resource).withStatus(ks).build();
+                                crdOperator.updateStatus(resourceWithNewStatus);
```

Comment:
So, the interesting point ... I think we should collect the status field and set it only at the end when the reconciliation succeeds. That might save some HTTP calls and performance. But it will also make the error handling easier IMHO.

Reply:
+1. Fixed